src/pidigits.rs

```diff
@@ -77,18 +62,21 @@ impl Context {
     }
 }
 impl Iterator for Context {
-    type Item = c_ulong;
-    fn next(&mut self) -> Option<c_ulong> {
+    type Item = u8;
+    fn next(&mut self) -> Option<u8> {
```

Comment:
Without the collect, this change is not so relevant. But I have no real optinion on that. As you want.